src/github.ts

```diff
@@ -1,12 +1,38 @@
 import * as httpm from '@actions/http-client';
+import {Octokit} from '@octokit/action';
+import * as semver from 'semver';
 
 export interface GitHubRelease {
   id: number;
   tag_name: string;
 }
 
 export const getRelease = async (version: string): Promise<GitHubRelease | null> => {
-  const url: string = `https://github.com/goreleaser/goreleaser/releases/${version}`;
+  const resolvedVersion: string = (await resolveVersion(version)) || version;
+  const url: string = `https://github.com/goreleaser/goreleaser/releases/${resolvedVersion}`;
   const http: httpm.HttpClient = new httpm.HttpClient('goreleaser-action');
   return (await http.getJson<GitHubRelease>(url)).result;
 };
+
+const resolveVersion = async (version: string): Promise<string | null> => {
+  if (!process.env.GITHUB_TOKEN) {
```

Comment:
I suppose this condition is in place to lift the restriction on the pagination of tags? I'm not a big fan of forcing the use of the GitHub token. I think it might be a good idea to have the list of tags available in JSON format on the goreleaser website. What do you think @caarlos0?

Reply:
SGTM, we can probably generate it at build time 🤔

Reply:
@caarlos0 Yeah we can do that easily with [`actions/github-script`](https://github.com/actions/github-script) to extract all releases and create a PR with something like that:

```yaml
    - name: Fetch releases
      uses: actions/github-script@v2
      with:
        github-token: ${{secrets.GITHUB_TOKEN}}
        script: |
          const fs = require("fs");
          const response = await github.paginate(
            github.repos.listReleases.endpoint.merge({
              ...context.repo
            })
          );
          fs.writeFileSync(`www/docs/static/releases.json`, JSON.stringify(response));
    -
      name: Create Pull Request
      uses: peter-evans/create-pull-request@v2
      with:
        token: ${{ secrets.GITHUB_TOKEN }}
        commit-message: "chore(ci): update releases metadata"
        title: "chore(ci): update releases metadata"
        body: |
          Releases metadata updated.
        branch: releases-metadata
        branch-suffix: timestamp
        labels: automerge
```

Reply:
DONE: https://goreleaser.com/static/releases.json